include/diaspora.php

```diff
@@ -161,6 +162,56 @@ private static function verify_magic_envelope($envelope) {
 	}
 
 	/**
+	 * @brief encrypts data via AES
+	 *
+	 * @param string $key The AES key
+	 * @param string $iv The IV (is used for CBC encoding)
+	 * @param string $data The data that is to be encrypted
+	 *
+	 * @return string encrypted data
+	 */
```

Comment:
You would get a huge performance improvement using something like:

    function AES256CBC_encrypt($data,$key,$iv) {
        return openssl_encrypt($data,'aes-256-cbc',str_pad($key,32,"\0"),OPENSSL_RAW_DATA,str_pad($iv,16,"\0"));
    }

    function AES256CBC_decrypt($data,$key,$iv) {
        return openssl_decrypt($data,'aes-256-cbc',str_pad($key,32,"\0"),OPENSSL_RAW_DATA,str_pad($iv,16,"\0"));
    }